tracing/src/main/java/com/palantir/tracing/Tracer.java

```diff
@@ -143,10 +141,12 @@ public static void fastCompleteSpan() {
      * Like {@link #fastCompleteSpan()}, but adds {@code metadata} to the current span being completed.
      */
     public static void fastCompleteSpan(Map<String, String> metadata) {
-        popCurrentSpan()
-                .filter(openSpan -> currentTrace.get().isObservable())
-                .map(openSpan -> toSpan(openSpan, metadata))
-                .ifPresent(Tracer::notifyObservers);
+        if (isTraceObservable()) {
```

Comment:
Why put this behind the `isTraceObservable` check?

Reply:
ack, this is a bug. fix incoming

Reply:
fixed